include/native/farm.h

```diff
@@ -57,21 +57,19 @@ void farm(parallel_execution_native & ex, Generator generate_op,
 {
   using namespace std;
   using result_type = typename result_of<Generator()>::type;
-  mpmc_queue<result_type> queue{ex.queue_size,ex.lockfree};
+  auto queue = ex.make_queue<result_type>();
 
   vector<thread> tasks;
   for (int i=0; i<ex.concurrency_degree(); ++i) {
     tasks.emplace_back([&](){
-      ex.register_thread();
+      auto manger = ex.thread_manager();
```

Comment:
Indeed, I change the name to manager.

It is not meant to be used. It is a RAII object. Consutructor acquires resource and destructor releases resource.